### PR TITLE
Tweaks to the new form csrf caching entry

### PR DESCRIPTION
**Pull request by weaverryan**
| Q | A |
| --- | --- |
| Doc fix? | no |
| New docs? | no |
| Applies to | all |
| Fixed tickets | n/a |

Hi guys!

After merging this nice new entry in #4141, I wanted to make a few minor language tweaks (the diff looks bigger than the changes really are). This included shortening a few sections and talking less about how _all_ reverse proxies refuse to cache pages with a session, because I don't (for example) believe this is true with Symfony's reverse proxy.

Thanks!

### CODE REVIEW

**Comment by dbu** [comment]
seems good to me, easier to read and no mistakes introduced as far as i can see.

File: cookbook/cache/form_csrf_caching.rst [hunk 1]
```diff
@@ -10,39 +10,33 @@ need to be cautious if you try to cache pages with forms including them.
...
-for each user.
-
-How to Cache Most of the Page and still Be Able to Use CSRF Protection
+Why Caching Pages with a CSRF token are Problematic
+---------------------------------------------------
+
+Typically, each user is assigned a unique CSRF token, which is stored in
+the session for validation. This means that if you *do* cache a page with
+a form containing a CSRF token, you'll cache the CSRF token of the *first*
+user only. When a user submits, the token won't match the token stored in
```
Comment (javiereguiluz) [inline cookbook/cache/form_csrf_caching.rst]: I think there are some missing words here:

```
When a user submits, the token ...
```

Proposal:

```
When a user submits the form, the token ...
```

File: cookbook/cache/form_csrf_caching.rst [hunk 2]
```diff
@@ -10,39 +10,33 @@ need to be cautious if you try to cache pages with forms including them.
...
-Varnish default behaviour is to not cache HTTP requests with cookies.
-
-If you think about it, if you managed to cache the form you would end up
-with many users getting the same token in the form generation. When these
-users try to send the form to the server, the CSRF validation will fail for
-them because the expected token is stored in their session and different
-for each user.
-
-How to Cache Most of the Page and still Be Able to Use CSRF Protection
+Why Caching Pages with a CSRF token are Problematic
```
Comment (xabbuh) [inline cookbook/cache/form_csrf_caching.rst]: Shouldn't it be "is" instead of "are"? And should we add a label for the old headline?

Comment (weaverryan) [inline cookbook/cache/form_csrf_caching.rst]: You're right! Fixed at sha: 36d1bacc487d0c16023742a9260444ab5d86e8bb

**Pull request merged**